### ISSUE TITLE
Change refersTo to distinguish between class or instance and different instance accesses

### ISSUE THREAD

**Issue by konradweiss** [Fraunhofer-AISEC/cpg#44]
We have to distinguis wheter or not a reference points to an instance of a class or the class itself. Also at some point when taking into consideration data flow analysis results we should try to distinguish between instances that are known to be different.
Currently I am considering multiedges or adding an intermediate node to transform ref --referseTo-->Class into ref --referseTo-->instance --instanceOf-->Class. 
Every instance will increase the graphs size so we should be conservative and not add node for every appearance of an unknown reference but rather have an edge or node representing all unknown references, preferably an edge, that prevents the code from assuming references pointing there use the same instance.

**Comment by oxisto** [comment]
Not sure if I understand the issue the correctly. Which "reference" are you referring to? Could you give an example of a reference to a class in the current CPG, ideally with code?

**Comment by konradweiss** [comment]
You were right with your skepticism, calls to static functions or members ar already recognized by the parser and we currently translate them into StaticCallExpressions and StaticCallExpressions and StaticReferenceDeclarations. Only the second proposal makes, therefore, sense, change them to distinguish different instance access of an Object. I can't reproduce it currently and I'm not certain if I mistakenly thought that I saw that behavior or it is because of another bug that I found.
